xcm/pallet-xcm-benchmarks/src/fungible/benchmarking.rs

```diff
@@ -0,0 +1,221 @@
+		let asset = T::get_multi_asset();
+		let assets: MultiAssets = vec![ asset.clone() ].into();
+		// this xcm doesn't use holding
+
+		let dest_location = T::valid_destination()?;
+		let dest_account = T::AccountIdConverter::convert(dest_location.clone()).unwrap();
+
+		<AssetTransactorOf<T>>::deposit_asset(&asset, &sender_location).unwrap();
+		assert!(T::TransactAsset::balance(&dest_account).is_zero());
+
+		let mut executor = new_executor::<T>(sender_location);
+		let instruction = Instruction::TransferAsset { assets, beneficiary: dest_location };
+		let xcm = Xcm(vec![instruction]);
+	}: {
+		executor.execute(xcm)?;
+	} verify {
+		assert!(T::TransactAsset::balance(&sender_account).is_zero());
+		assert!(!T::TransactAsset::balance(&dest_account).is_zero());
+	}
+
+	transfer_reserve_asset {
+		let (sender_account, sender_location) = account_and_location::<T>(1);
+		let dest_location = T::valid_destination()?;
+		let dest_account = T::AccountIdConverter::convert(dest_location.clone()).unwrap();
+
+		let asset = T::get_multi_asset();
+		<AssetTransactorOf<T>>::deposit_asset(&asset, &sender_location).unwrap();
+		let assets: MultiAssets = vec![ asset ].into();
+		assert!(T::TransactAsset::balance(&dest_account).is_zero());
+
+		let mut executor = new_executor::<T>(sender_location);
+		let instruction = Instruction::TransferReserveAsset {
+			assets,
+			dest: dest_location,
+			xcm: Xcm::new()
+		};
+		let xcm = Xcm(vec![instruction]);
+	}: {
+		executor.execute(xcm)?;
+	} verify {
+		assert!(T::TransactAsset::balance(&sender_account).is_zero());
+		assert!(!T::TransactAsset::balance(&dest_account).is_zero());
+		// TODO: Check sender queue is not empty.
+	}
+
+	receive_teleported_asset {
+		// If there is no trusted teleporter, then we skip this benchmark.
+		let (trusted_teleporter, teleportable_asset) = T::TrustedTeleporter::get().ok_or(
+			BenchmarkError::Override(
+				BenchmarkResult::from_weight(T::BlockWeights::get().max_block)
+			)
+		)?;
+
+		let assets: MultiAssets = vec![ teleportable_asset ].into();
+
+		let mut executor = new_executor::<T>(trusted_teleporter);
+		let instruction = Instruction::ReceiveTeleportedAsset(assets.clone());
+		let xcm = Xcm(vec![instruction]);
+	}: {
+		executor.execute(xcm).map_err(|_| {
+			BenchmarkError::Override(
+				BenchmarkResult::from_weight(T::BlockWeights::get().max_block)
+			)
+		})?;
+	} verify {
+		assert!(executor.holding.ensure_contains(&assets).is_ok());
+	}
+
+	deposit_asset {
+		let asset = T::get_multi_asset();
+		let mut holding = worst_case_holding();
+
+		// Add our asset to the holding.
+		holding.subsume(asset.clone());
+
+		// our dest must have no balance initially.
+		let dest_location = T::valid_destination()?;
+		let dest_account = T::AccountIdConverter::convert(dest_location.clone()).unwrap();
+		assert!(T::TransactAsset::balance(&dest_account).is_zero());
+
+		let mut executor = new_executor::<T>(Default::default());
```

Comment:
nit: might want to make it `fn new_executor<T: Config>(_: MultiLocation, initial_holding: Assets)`